### PR TITLE
Allowing the TS task's configuration to be extended on an instance-by-instance basis.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: core-build/gulp-core-build-typescript/src/TypeScriptTask.ts [hunk 1]
```diff
@@ -68,6 +69,11 @@ export interface ITypeScriptTaskConfig {
    * If defined, drop typescript files build using modules=ES6.
    */
   libES6Dir?: string;
+
+  /**
+   * If defined, apply these settings on top of the standardized project TypeScript compiler configuration.
+   */
+  configAddons?: ts.Settings;
```
Comment (nickpape-msft) [inline core-build/gulp-core-build-typescript/src/TypeScriptTask.ts]: use full words e.g. "configurationAddons" ?

**Comment by nickpape-msft** [comment]
approved

**Comment by iclanton** [comment]
Approved.

**Comment by iclanton** [comment]
Approved.

**Pull request merged**